src/software/geom/algorithms/contains.cpp

```diff
@@ -61,13 +62,17 @@ bool contains(const Polygon& container, const Point& contained)
 bool contains(const Ray& container, const Point& contained)
 {
     Point point_in_ray_direction = container.getStart() + container.toUnitVector();
+    bool point_is_ray_start      = contained == container.getStart();
 
-    bool point_is_ray_start = contained == container.getStart();
+    // We use broader values for epsilon and ulps because the default values brought
+    // warnings from simulated tests
```

Comment:
We shouldn't be designing/tuning contains around how simulated test warnings. I think fixing this in a non-hacky way is pretty tricky, so please revert all the changes in this file and we'll deal with them later.

Reply:
We could make epsilon and ulps default parameters that can be overwritten, like one version of contains does?

Reply:
I don't see how you could pass these values all the way down. In any case, I've looked into this and there are actual bugs in the contains function that can affect not just these warnings but also calcbestshot

Reply:
fair enough, ill revert the changes and let them be addressed at another time

Reply:
removed changes affecting contains.cpp